sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -130,7 +126,16 @@ object FunctionRegistry {
     expression[Last]("last"),
     expression[Max]("max"),
     expression[Min]("min"),
-    expression[Sum]("sum")
+    expression[Sum]("sum"),
+
+    // string functions
+    expression[Upper]("lcase"),
+    expression[Lower]("lower"),
+    expression[StringLength]("strlen"),
```

Comment:
don't rename this one since we need hive compatibility here... only rename the data frame function.